Content.Server/Shipyard/Systems/ShipyardSystem.cs

```diff
@@ -0,0 +1,178 @@
+using Content.Server.Shuttles.Systems;
+using Content.Server.Shuttles.Components;
+using Content.Server.Station.Components;
+using Content.Server.Cargo.Systems;
+using Content.Server.Station.Systems;
+using Content.Shared.Shipyard;
+using Content.Server.Shipyard.Components;
+using Content.Shared.Mobs.Components;
+using Content.Shared.GameTicking;
+using Robust.Server.GameObjects;
+using Robust.Server.Maps;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Content.Shared.CCVar;
+using Robust.Shared.Configuration;
+using Content.Server.Cargo.Components;
+
+namespace Content.Server.Shipyard.Systems
+{
+
+    public sealed partial class ShipyardSystem : SharedShipyardSystem
+    {
+        [Dependency] private readonly IConfigurationManager _configManager = default!;
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly PricingSystem _pricing = default!;
+        [Dependency] private readonly ShuttleSystem _shuttle = default!;
+        [Dependency] private readonly StationSystem _station = default!;
+        [Dependency] private readonly MapLoaderSystem _map = default!;
+        [Dependency] private readonly ShipyardConsoleSystem _shipyardConsole = default!;
+
+        public MapId? ShipyardMap { get; private set; }
+        private float _shuttleIndex;
+        private const float ShuttleSpawnBuffer = 1f;
+        private ISawmill _sawmill = default!;
+        private bool _enabled;
+
+        public override void Initialize()
+        {
+            _enabled = _configManager.GetCVar(CCVars.Shipyard);
+            _configManager.OnValueChanged(CCVars.Shipyard, SetShipyardEnabled);
+            _sawmill = Logger.GetSawmill("shipyard");
+            _shipyardConsole.InitializeConsole();
+            SubscribeLocalEvent<ShipyardConsoleComponent, ComponentInit>(OnShipyardStartup);
+            SubscribeLocalEvent<RoundRestartCleanupEvent>(OnRoundRestart);
+        }
+
+        private void OnShipyardStartup(EntityUid uid, ShipyardConsoleComponent component, ComponentInit args)
+        {
+            if (!_enabled)
+                return;
+
+            SetupShipyard();
+        }
+
+        private void OnRoundRestart(RoundRestartCleanupEvent ev)
+        {
+            _configManager.UnsubValueChanged(CCVars.Shipyard, SetShipyardEnabled);
+            CleanupShipyard();
+        }
```

Comment:
Is this unsub supposed to be here? Seems like it should be on system shutdown instead.